.github/workflows/ci.yml

```diff
@@ -0,0 +1,564 @@
+      - uses: Alfresco/acs-community-packaging/.github/actions/get-build-info@feature/ACS-3844_migrate-to-GHA
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.25.0
+      - name: "Init"
+        run: bash ./scripts/ci/init.sh
+      - name: "Set up the environment"
+        run: docker-compose -f ./scripts/ci/docker-compose/docker-compose.yaml --profile activemq up -d
+      - name: "Run tests"
+        run: mvn -B test -pl repository -am -Dtest=CamelRoutesTest,CamelComponentsTest -DfailIfNoTests=false
+      - name: "Clean Maven cache"
+        if: ${{ always() }}
+        run: bash ./scripts/ci/cleanup_cache.sh
+
+  repository_app_context_test_suites:
+    name: Repository - ${{ matrix.testSuite }}
+    runs-on: ubuntu-latest
+    needs: [prepare]
+    if: >
+      ! contains(github.event.head_commit.message, '[skip repo]') &&
+      ! contains(github.event.head_commit.message, '[skip tests]') &&
+      ! contains(github.event.head_commit.message, '[force]')
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - testSuite: AppContext01TestSuite
+            compose-profile: with-transform-core-aio
+          - testSuite: AppContext02TestSuite
+            compose-profile: default
+          - testSuite: AppContext03TestSuite
+            compose-profile: with-transform-core-aio
+          - testSuite: AppContext04TestSuite
+            compose-profile: with-transform-core-aio
+          - testSuite: AppContext05TestSuite
+            compose-profile: default
+            mvn-options: '"-Didentity-service.auth-server-url=http://${HOST_IP}:8999/auth"'
+          - testSuite: AppContext06TestSuite
+            compose-profile: with-transform-core-aio
+          - testSuite: AppContextExtraTestSuite
+            compose-profile: with-transform-core-aio
+          - testSuite: MiscContextTestSuite
+            compose-profile: with-transform-core-aio
+          - testSuite: SearchTestSuite
+            compose-profile: default
+            mvn-options: '-Dindex.subsystem.name=solr6'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/acs-community-packaging/.github/actions/get-build-info@feature/ACS-3844_migrate-to-GHA
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.25.0
+      - name: "Init"
+        run: bash ./scripts/ci/init.sh
+      - name: "Set up IDS"
+        run: |
+          if [ "${{ matrix.testSuite }}" == 'AppContext05TestSuite' ];
+          then
+            mkdir -p "${HOME}/tmp"
+            cp repository/src/test/resources/realms/alfresco-realm.json "${HOME}/tmp"
+            echo "HOST_IP=$(hostname  -I | cut -f1 -d' ')" >> $GITHUB_ENV
+            docker run -d -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin -e DB_VENDOR=h2 -p 8999:8080 -e KEYCLOAK_IMPORT=/tmp/alfresco-realm.json -v $HOME/tmp/alfresco-realm.json:/tmp/alfresco-realm.json alfresco/alfresco-identity-service:1.2
+          fi
```

Comment:
move into a `AppContext05TestSuite-setup.sh` script and here check if `${{ matrix.testSuite }}-setup.sh` script exists and execute it

Reply:
Nice solution 👍  Done.